issues/user-input.ts

```diff
@@ -1,17 +1,30 @@
-interface VerySimpleDatabase {
+
+// Find out why our error database is filling with following error when using savePassword/Age in user-input module:
+// 'value' is not a string or number (VerySimpleDatabase.insert)
+// Also, double-check that the functions are otherwise correct.
+
+// Write unit tests for the functions, there don't seem to be any at the moment.
+
+//Initial thoughts --> type issue being created 
+//could be trying submit a null or undefined value
+//parseINt could be trying to  submit a null or undefined value 
+
+export interface VerySimpleDatabase {
+
   insert(key: string, value: string | number): void;
 }
 
 export function savePassword(
   database: VerySimpleDatabase,
-  httpPostParams: Record<string, string>
+  httpPostParams: {password: string}
```

Comment:
You cannot generally trust in something like `httpPostParams` because the HTTP request may contain any parameters. So instead you should check the types of the parameters and then fail if the type is not correct.

It might also be a good idea to not save the password in readable form, so there's also a security issue here.